untitled/src/Module7/homework/Task3/Main2.java

```diff
@@ -0,0 +1,89 @@
+package Module7.homework.Task3;
+
+import Module7.homework.Task2.Currency;
+
+import java.util.*;
+
+/**
+ * Created by user on 03.10.2016.
+ */
+public class Main2 {
+
+    static Set<Order1> order1Set = new TreeSet<Order1>(new Comparator<Order1>() {
+        @Override
+        public int compare(Order1 o1, Order1 o2) {
+            return o2.getPrice() - o1.getPrice();
+        }
+    });
+    static Iterator<Order1> iterator = order1Set.iterator();
+
+    //----------------------------------------Check if Set contain Order where Users lastName is - Petrov----------------
+
+    public static void searchingLastName(Set<Order1> order1Set, String lastName) {
+        System.out.println("********** Check if Set contain Order where User�s lastName is - Petrov*************");
+        boolean flag = false;
+        for (Order1 o : order1Set) {
+
+            //System.out.println(o);
+            if (o.getUser1().getLastName() == lastName) {
```

Comment:
You shoudl never compare Strings with '==' because String is  an object and you check their hashcodes. Use equals() instead.